ouroboros-consensus/src/Ouroboros/Consensus/Mempool/API.hs

```diff
@@ -98,12 +103,38 @@ data Mempool m blk = Mempool {
       -- they have already been included. (Distinguishing between these two
       -- cases can be done in theory, but it is expensive unless we have an
       -- index of transaction hashes that have been included on the blockchain.)
-      addTxs :: [GenTx blk] -> m [(GenTx blk, ApplyTxErr blk)]
+      addTxs :: [(GenTxId blk, GenTx blk)] -> m [(GenTx blk, ApplyTxErr blk)]
 
-      -- | Get all transactions in the mempool (oldest to newest)
+      -- | Sync the transactions in the mempool with the ledger state of the
+      -- 'ChainDB'. Invalid transactions will be returned along with the
+      -- validation error.
+    , syncState :: STM m [(GenTx blk, ApplyTxErr blk)]
+
+      -- | Get all transactions in the mempool along with their associated
+      -- ticket numbers (oldest to newest).
       --
-      -- Guarantees that these transactions will be valid (when applied strictly
-      -- in order) with respect to a call to 'getLedgerState' run /in the same
+      -- n.b. This function provides /no guarantee/ that the resulting
+      -- transactions will be valid with respect to the ledger state of the
+      -- 'ChainDB'. However, it is guaranteed that these transactions will be
+      -- valid (with respect to the ledger state) if this function is called
+      -- immediately following a call to 'syncState', /within the same
       -- transaction/.
-    , getTxs :: STM m (Seq (GenTx blk))
+    , getTxs :: STM m [(GenTxId blk, GenTx blk, idx)]
+
+      -- | Get all transactions in the mempool, along with their associated
+      -- ticket numbers, which are associated with a ticket number greater
+      -- than the one provided.
+    , getTxsAfter :: idx -> STM m [(GenTxId blk, GenTx blk, idx)]
+
+      -- | Get a specific transaction from the mempool by its ticket number,
+      -- if it exists.
+    , getTx :: idx -> STM m (Maybe (GenTxId blk, GenTx blk))
+
+      -- | Get a specific transaction from the mempool by its identifier, if
+      -- it exists.
+    , getTxById :: GenTxId blk -> STM m (Maybe (GenTx blk))
```

Comment:
What do we need this function for? We're not using it in this PR, and I can't immediately think what we need it for.

It'd simplify the impl by removing the need for the `Map` and keeping it in sync.